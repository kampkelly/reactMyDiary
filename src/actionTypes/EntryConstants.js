export const NEW_ENTRY = 'NEW_ENTRY';
