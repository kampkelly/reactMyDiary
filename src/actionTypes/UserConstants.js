export const SIGNIN = 'SIGNIN';
export const SIGNUP = 'SIGNUP';
export const SIGNOUT = 'SIGNOUT';
export const PROFILE = 'PROFILE';
export const NOTIFICATION = 'NOTIFICATION';