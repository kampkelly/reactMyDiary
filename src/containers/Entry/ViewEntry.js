import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './ViewEntry.scss';
import '../../styles/sidebar.scss';
import { ShowEntry, DeleteEntry } from '../../requests/EntryRequests';
import { checkNotice } from '../../helpers/checkNotice';

/**
 * @class ViewEntry
 * @extends {Component}
 */
export class ViewEntry extends Component {
  /**
   *Creates an instance of NewEntry.
   * @memberof ViewEntry
   */
  constructor() {
    super();
    this.state = {
      entry: {},
      mmessage: '',
      success: false
    };
    this.deleteEntry = this.deleteEntry.bind(this);
  }

  componentDidMount() {
    document.querySelector('main').style.backgroundImage = "url('https://i.imgur.com/n4ttyU5.jpg')";
    checkNotice();
    this.props.ShowEntry(this.props.match.params.id);
    document.querySelector('body').insertAdjacentHTML('afterbegin', `<img src="http://res.cloudinary.com/ddfepbdqg/image/upload/v1543597369/Rolling.svg" id="loading"></img>`);
  }

  deleteEntry(e) {
    e.preventDefault();
    this.props.DeleteEntry(this.state.entry.id);
  }

  /**
   * @description - This method runs whenever redux state changes
   * @returns {object} state
   * @param {object} props
   * @param {object} state
   * @memberof ViewTag
   */
  static getDerivedStateFromProps(props, state) {
    return {
      entry: props.entry,
      message: props.message,
      success: props.success
    };
  }

  /**
   *
   * @returns {jsx} - jsx
   * @memberof NewEntry
   */
  viewEntry(e) {
    if (this.state.success && this.state.entry.createdat) {
      const date = this.state.entry.createdat.split('T')[0];
      return (<article><h3 className="title text-center white-text">{this.state.entry.title}</h3><span className="small-text light-text primary-text"><b>{date}</b></span><div><small><Link key={this.state.entry.id} to={`/entries/${this.state.entry.id}/update`}>Update</Link> <a href="delete" className="danger-text delete-entry" onClick={this.deleteEntry}>Delete</a></small></div><p className="description white-text">{this.state.entry.description}</p></article>);
    } else {
      return (<h3 className="text-center white-text">{this.state.message}</h3>);
    }
  }

  /**
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   * @memberof Header
   */
  render() {
    return (
      <div>
        <section id="show">
          {this.viewEntry()}
        </section>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  entry: state.entry.entry,
  message: state.entry.message,
  success: state.entry.success
});

ViewEntry.propTypes = {
  match: PropTypes.object,
  message: PropTypes.string,
  DeleteEntry: PropTypes.func,
  ShowEntry: PropTypes.func,
  success: PropTypes.bool
};

export default connect(mapStateToProps, {
  DeleteEntry, ShowEntry
})(ViewEntry);

