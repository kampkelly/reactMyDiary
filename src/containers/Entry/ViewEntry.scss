$primary-color: #137ad4;
$secondary-color: #5E382B;
$white-color: #fafafa;
$smartphone: "(min-width: 0px) and (max-width: 599px)";
$tablet: "(min-width: 600px) and (max-width: 900px)";

main {
	// background-image: url("../../assets/2.jpg");
  background-size: cover;
	background-position: center;
	background-repeat-x: no-repeat;
  background-repeat-y: no-repeat;
}
#show {
  height: 100%;
  width: 70%;
	margin: auto;
  padding: 10px;
  padding-top: 60px;
    article {
        div {
            float: right;
            a:first-of-type {
                padding-right: 10px;
				color: #0d529d;
				transition: 1s;
            }
        }
        .title {
            font-size: 1.4em;
        }
        .description {
            line-height: 33px;
            text-align: justify;
            font-size: 1em;
            clear: both;
        }
    }
}
@media #{$smartphone} {
	#show {
		width: 90%;
		margin-top: 10px;
	}
}
@media #{$tablet} {
	#show {
		width: 90%;
		margin-top: 20px;
	}
}
