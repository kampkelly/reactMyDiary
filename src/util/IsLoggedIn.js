export const isLoggedIn = () => {
  if (localStorage.getItem('diary_token') !== null) {
    return true;
  }
  return false;
};
