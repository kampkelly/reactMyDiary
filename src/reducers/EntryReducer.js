import { NEW_ENTRY, SHOW_ENTRY } from '../actionTypes/EntryConstants';
import { asyncActionName } from '../util/AsyncUtil';


const initialState = {
  entries: [],
  entry: {},
  loading: false,
  message: '',
  success: false,
  failure: false,
};

const entryReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case asyncActionName(NEW_ENTRY).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(NEW_ENTRY).success:
      return {
        ...state,
        success: action.payload
      };
    case asyncActionName(NEW_ENTRY).failure:
      document.getElementById('loading').style.display = 'none';
      document.querySelector('.form_error_text').style.display = 'block';
      document.querySelector('.form_error_text small').textContent = action.payload.error;
      return {
        ...state,
      };
    case asyncActionName(SHOW_ENTRY).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(SHOW_ENTRY).success:
      return {
        ...state,
        entry: action.payload, success: true
      };
    case asyncActionName(SHOW_ENTRY).failure:
      return {
        ...state, message: action.payload.error
      };
    default:
      return state;
  }
};

export default entryReducer;