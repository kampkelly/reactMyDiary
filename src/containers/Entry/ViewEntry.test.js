import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ViewEntry from './ViewEntry';
import EntryReducer from '../../reducers/EntryReducer';
import { asyncActions } from '../../util/AsyncUtil';
import { SHOW_ENTRY } from '../../actionTypes/EntryConstants';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  entry: {
    entry: {},
    message: '',
    success: false
  }
});

const props = {
  match: {
    params: {
      id: 1
    }
  },
  ShowEntry: jest.fn()
};

let component;
let myComponent;

describe('<ViewEntry/>', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <ViewEntry {...props} />
      </Provider>
    );
    // myComponent = component.dive({ context: { store } }).dive();
    document.body.innerHTML =
    '<body>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</body>';
  });
  it('should show a new entry', () => {
    const payload = true;
    const expectedAction = {
      type: 'SHOW_ENTRY_SUCCESS',
      payload
    };
    expect(asyncActions(SHOW_ENTRY).success(payload)).toEqual(expectedAction);
  });
  it('should dispatch an action when getting an entry is loading', () => {
    const payload = false;
    const action = {
      type: 'SHOW_ENTRY_LOADING',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({
      loading: payload
    });
  });
  it('should dispatch an action to get an entry', () => {
    const payload = {};
    const action = {
      type: 'SHOW_ENTRY_SUCCESS',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({
      entry: payload,
      success: true,
    });
  });
  it('should dispatch an action when getting an entry fails', () => {
    const payload = {
      error: ''
    };
    const action = {
      type: 'SHOW_ENTRY_FAILING',
      payload
    };
    const newState = EntryReducer({}, action);
    expect(newState).toEqual({
      message: payload.error
    });
  });
});
