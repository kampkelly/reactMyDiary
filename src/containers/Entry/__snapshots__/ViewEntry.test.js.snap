// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewEntry/> should render without throwing an error 1`] = `
<Connect(ViewEntry)
  ShowEntry={[MockFunction]}
  match={
    Object {
      "params": Object {
        "id": 1,
      },
    }
  }
/>
`;
